examples/uploader/_localrepo.py

```diff
@@ -0,0 +1,133 @@
+        raise NotImplementedError  # we never call snapshot
+
+    @property
+    def snapshot_info(self) -> MetaFile:
+        raise NotImplementedError  # we never call timestamp
+
+    def open(self, role: str) -> Metadata:
+        """Return cached (or fetched) metadata"""
+
+        # if there is a metadata version fetched from remote, use that
+        # HACK: access Updater internals
+        # pylint: disable=protected-access
+        if role in self.updater._trusted_set:
+            return copy.deepcopy(self.updater._trusted_set[role])
+
+        # otherwise we're creating metadata from scratch
+        md = Metadata(Targets())
+        # this makes version bumping in close() simpler
+        md.signed.version = 0
+        return md
+
+    def close(self, role: str, md: Metadata) -> None:
+        """Store a version of metadata. Handle version bumps, expiry, signing"""
+        md.signed.version += 1
+        md.signed.expires = datetime.utcnow() + self.expiry_period
+
+        with open(f"{self.key_dir}/{role}", "rt", encoding="utf-8") as f:
+            signer = SSlibSigner(json.loads(f.read()))
+
+        md.sign(signer, append=False)
+
+        # Upload using "api/role"
+        uri = f"{self.base_url}/api/role/{role}"
+        r = requests.post(uri, data=md.to_bytes(JSONSerializer()), timeout=5)
+        r.raise_for_status()
+
+    def add_target(self, role: str, targetpath: str) -> bool:
+        """Add target to roles metadata and submit new metadata version"""
+
+        # HACK: make sure we have the roles metadata in updater._trusted_set
+        # (or that we're publishing the first version)
+        try:
+            self.updater.get_targetinfo(targetpath)
+        except RepositoryError:
+            # HACK Assume this is because we're just publishing version 1
+            # (so the roles metadata does not exist on server yet)
+            pass
+
+        data = bytes(targetpath, "utf-8")
+        targetfile = TargetFile.from_data(targetpath, data)
+        try:
+            with self.edit(role) as delegated:
+                delegated.targets[targetpath] = targetfile
+
+        except Exception as e:  # pylint: disable=broad-except
+            print(f"Failed to submit new {role} with added target: {e}")
+            return False
+
+        print(f"Uploaded role {role} v{delegated.version}")
+        return True
+
+    def add_delegation(self, role: str) -> bool:
+        """Use the (unauthenticated) delegation adding API endpoint"""
+        keydict = keys.generate_ed25519_key()
+        pubkey = SSlibKey.from_securesystemslib_key(keydict)
+
+        data = {pubkey.keyid: pubkey.to_dict()}
+        url = f"{self.base_url}/api/delegation/{role}"
+        r = requests.post(url, data=json.dumps(data), timeout=5)
+        if r.status_code != 200:
+            print(f"delegation failed with {r}")
+            return False
+
+        # TODO: Once the Signer API is ready, use the priv key uri, store the file encrypted etc
```

Comment:
I think the Signer API is ready, right?

Reply:
Yes and encrypting with passphrase would make it clearer to user when the key is created and when it's being used... but it would be more code and the example is quite large already so I'll just remove the TODO